spec/presenters/two_factor_authentication/selection_presenter_spec.rb

```diff
@@ -22,7 +22,73 @@ def method
   end
 
   describe '#disabled?' do
-    it { expect(presenter.disabled?).to eq(false) }
+    let(:single_configuration_only) {}
+    let(:mfa_configuration_count) {}
+
+    before do
+      allow(presenter).to receive(:single_configuration_only?).and_return(single_configuration_only)
+      allow(presenter).to receive(:mfa_configuration_count).and_return(mfa_configuration_count)
+    end
+
+    context 'without single configuration restriction' do
+      let(:single_configuration_only) { false }
+
+      it 'is an mfa that allows multiple configurations' do
+        expect(presenter.disabled?).to eq(false)
+      end
+    end
+
+    context 'with single configuration only' do
+      let(:single_configuration_only) { false }
+
+      context 'with unimplemented mfa count (nil)' do
+        let(:mfa_configuration_count) { nil }
+
+        it 'is mfa with unimplemented mfa count and single config' do
+          expect(presenter.disabled?).to eq(false)
+        end
+      end
+
+      context 'with no configured mfas' do
+        let(:mfa_configuration_count) { 0 }
+
+        it 'is configured with no mfa' do
+          expect(presenter.disabled?).to eq(false)
+        end
+      end
+
+      context 'with at least one configured mfa' do
+        let(:mfa_configuration_count) { 1 }
+
+        it 'is mfa with at least one configured' do
+          expect(presenter.disabled?).to eq(false)
```

Comment:
There was a typo in my original suggestion. Since we want to test 'single configuration only' in this path, we should set the `single_configuration_only` value to `true`.

```suggestion
    context 'with single configuration only' do
      let(:single_configuration_only) { true }

      context 'with unimplemented mfa count (nil)' do
        let(:mfa_configuration_count) { nil }

        it 'is mfa with unimplemented mfa count and single config' do
          expect(presenter.disabled?).to eq(false)
        end
      end

      context 'with no configured mfas' do
        let(:mfa_configuration_count) { 0 }

        it 'is configured with no mfa' do
          expect(presenter.disabled?).to eq(false)
        end
      end

      context 'with at least one configured mfa' do
        let(:mfa_configuration_count) { 1 }

        it 'is mfa with at least one configured' do
          expect(presenter.disabled?).to eq(true)
```

Reply:
I think the above changes will also reveal a potential bug with `nil` handling like I raised at https://github.com/18F/identity-idp/pull/8863#discussion_r1290081760 .